### DISCUSSION TITLE
Issues configuring obsidian.nvim

### DISCUSSION THREAD

**Discussion by pete-dot-m** [epwalsh/obsidian.nvim#401] [Q&A]
Hello!
I'll apologize in advance as this may be a bit long, but I wanted to include as much information as possible...
First off:
Pop! Os latest (22.04 w/ kernel 6.6.10-76060610)
I3 v4.23
NeoVim  v0.9.5 using Lazy for plugin management
obsidian.nvim  v3.2.0
Alacritty v0.14.0-dev
Obsidian 1.5.3

I have my Obsidian vault configured as "myvault" stored in ~/vaults/myvault and have configured the Git plugin to sync to without issue. Everything from the Obsidian side works as expected.

My issues come up when trying to get nvim/obsidian.nvim up and running. As stated above, I'm using Lazy for package management, and I've got my individual configs split out into separate .lua files, same as [omerxx' dotfiles](https://github.com/omerxx/dotfiles/tree/master/nvim)

```shell
pete@pop-os: ~
$ tree vaults                                                                                                                           [8:21:53]
vaults
└── myvault
    ├── 0.Inbox
    │   ├── 2024-02-13.md
    │   └── something.md
    ├── 1.Projects
    │   ├── City Service.md
    │   └── NLWO.md
    ├── 2.Areas
    ├── 3.Resources
    │   └── 2nd Brain.md
    ├── 4.Archive
    ├── My Areas.md
    ├── My Projects.md
    ├── My Resources.md
    └── Templates
        └── Resource Template.md

7 directories, 9 files
```

I add the reference to my vault in my obsidian.lua file:

obsidian.lua...
```lua
require("obsidian").setup({
    workspaces = {
      {
         name = "myvault",
         path = "~/vaults/myvault",
      },
    },
    completion = {
       -- Set to false to disable completion.
      nvim_cmp = true,
    ...
```

But no matter how I name my vault and/or directory structure, I always get the following error when starting up nvim:

![the-error](https://github.com/epwalsh/obsidian.nvim/assets/434857/dd8d1786-2ba4-4809-b7c9-89be64bd51b7)

For those interested, I've captured the rest of my relevant setup:

init.lua...
```lua
...
require('plugins.obsidian')
...
```

lazy.lua...
```lua
...
 {
        "epwalsh/obsidian.nvim", 
         version = "*",  -- recommended, use latest release instead of latest commit
         lazy = true,
         ft = "markdown",
         dependencies = {
          "nvim-lua/plenary.nvim",
        },
        opts = {
        },  
},
...
```

Any suggestions would be greatly appreciated!

Thanks, pete-dot-m!

**Comment by epwalsh** [comment]
Hey @pete-dot-m, the first snippet looks fine:

```lua
require("obsidian").setup({
    workspaces = {
      {
         name = "myvault",
         path = "~/vaults/myvault",
      },
    },
    completion = {
       -- Set to false to disable completion.
      nvim_cmp = true,
    ...
```

But the 2nd snippet does not:

```lua
...
 {
        "epwalsh/obsidian.nvim", 
         version = "*",  -- recommended, use latest release instead of latest commit
         lazy = true,
         ft = "markdown",
         dependencies = {
          "nvim-lua/plenary.nvim",
        },
        opts = {
        },  
},
...
```

The problem with the 2nd snippet is that `opts` is an empty table, so no workspaces are defined, which would lead to that error.

**Reply by epwalsh** [reply]
Generally you want to let `lazy` call the plugin's setup method (like it does in the 2nd snippet) instead of calling it directly (like in your first snippet. But you definitely don't want both. So I would remove the first snippet, wherever that is, and change the 2nd to this:

```lua
...
 {
       "epwalsh/obsidian.nvim", 
        version = "*",  -- recommended, use latest release instead of latest commit
        lazy = true,
        ft = "markdown",
        dependencies = {
          "nvim-lua/plenary.nvim",
        },
        opts = {
          workspaces = { .. },
          ...
        },  
},
...
```

**Reply by pete-dot-m** [reply]
Ok, thanks, I'll give that a try.
I should have pointed out that lazy's init.lua is called before obsidian's, but I'll try it regardless and report back.

Thanks!

**Reply by pete-dot-m** [reply]
That seems to have fixed it. What I ended up doing was remove the empty opts table from the lazy init.lua and kept all the plug-in specific stuff in the obsidian.lua file. 

Thanks for the quick response!

Pete
